Userland/Applications/PackageManager/MarkdownTableFinder.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2023, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/String.h>
+#include <AK/StringView.h>
+#include <AK/Types.h>
+#include <LibMarkdown/Document.h>
+#include <LibMarkdown/Visitor.h>
+
+class MarkdownTableFinder final : Markdown::Visitor {
+public:
+    ~MarkdownTableFinder() = default;
+
+    static MarkdownTableFinder analyze(Markdown::Document const& document)
+    {
+        MarkdownTableFinder finder;
+        document.walk(finder);
+        return finder;
+    }
+
+    size_t table_count() const { return m_tables.size(); }
+    Vector<Markdown::Table const*> const& tables() const { return m_tables; }
+
+private:
+    MarkdownTableFinder() { }
+
+    virtual RecursionDecision visit(Markdown::Table const& table) override
+    {
+        m_tables.append(&table);
+        if (m_tables.size() > 1)
+            return RecursionDecision::Break;
```

Comment:
This looks oddly specific. Why are we only looking for the first two tables?

Reply:
We look for the first table only, I will fix it now.